### PR TITLE
WS subscription modeluri issues (Fix #82, Fix #87)

### LINKED ISSUE #82: WS subscription with absolute path do not work

### LINKED ISSUE #87: WS subscription should return error in case of empty subscription path

### PR DESCRIPTION
**Pull request by ndoschek**
- Fix WS subscription on absolute paths
- Allows WS subscriptions only for valid modeluris
- Update UnitTest cases for subscribe/unsubscribe
- Fix a typo in ModelServerModule
- Remove outdated comment in SessionController

### CODE REVIEW

File: com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/ModelServerRouting.java [hunk 1]
```diff
@@ -114,16 +116,25 @@ public void bindRoutes() {
...
-					wsHandler.onClose(ctx -> getController(SessionController.class).unsubscribe(ctx));
-					wsHandler.onError(ctx -> {});
-					wsHandler.onMessage(ctx -> {});
+					wsHandler.onConnect(ctx -> {
+						getQueryParam(ctx.queryParamMap(), "modeluri")
+							.map(this::adaptModelUri)
+							.ifPresentOrElse(
+									modeluri -> {
+										if (!getController(SessionController.class).subscribe(ctx, modeluri)) {
+											handleWsErrorAndCloseSession(ctx, String.format("Cannot subscribe to '%s': modeluri is no valid model resource", modeluri));
```
Comment (cdamus) [inline com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/ModelServerRouting.java]:
```suggestion
											handleWsErrorAndCloseSession(ctx, String.format("Cannot subscribe to '%s': modeluri is not a valid model resource", modeluri));
```

**Review by cdamus — APPROVED** [review]
💯

File: com.eclipsesource.modelserver.emf/src/test/java/com/eclipsesource/modelserver/emf/common/SessionControllerTest.java [hunk 2]
```diff
@@ -67,24 +69,37 @@
...
 
-		assertFalse(sessionController.isClientSubscribed(clientCtx));
-
-		sessionController.subscribe(clientCtx, clientCtx.pathParam("modeluri"));
-
+		assertTrue(sessionController.subscribe(clientCtx, clientCtx.pathParam("modeluri")));
 		assertTrue(sessionController.isClientSubscribed(clientCtx));
+		verify(clientCtx).send(argThat(jsonNodeThat(containsRegex("(?i)\"type\":\"success\""))));
+
+		// try to subscribe to an invalid modeluri
```
Comment (edgarmueller) [inline com.eclipsesource.modelserver.emf/src/test/java/com/eclipsesource/modelserver/emf/common/SessionControllerTest.java:83]: Could we extract this into a test of its own?

File: com.eclipsesource.modelserver.emf/src/test/java/com/eclipsesource/modelserver/emf/common/SessionControllerTest.java [hunk 3]
```diff
@@ -67,24 +69,37 @@
...
+		//
+		when(clientCtx2.pathParam("modeluri")).thenReturn("tedioustesturi");
+		assertFalse(sessionController.subscribe(clientCtx2, clientCtx2.pathParam("modeluri")));
+		assertFalse(sessionController.isClientSubscribed(clientCtx2));
 	}
 
 	@Test
 	public void testUnsubscribe() {
 		testSubscribe();
```
Comment (edgarmueller) [inline com.eclipsesource.modelserver.emf/src/test/java/com/eclipsesource/modelserver/emf/common/SessionControllerTest.java]: I'd prefer if we wouldn't call another test directly here, extract method instead.

File: com.eclipsesource.modelserver.emf/src/test/java/com/eclipsesource/modelserver/emf/common/SessionControllerTest.java [hunk 4]
```diff
@@ -67,24 +69,37 @@
...
 		assertTrue(sessionController.isClientSubscribed(clientCtx));
-
-		sessionController.unsubscribe(clientCtx);
-
+		assertTrue(sessionController.unsubscribe(clientCtx));
+		assertFalse(sessionController.isClientSubscribed(clientCtx));
+		
+		// try to unsubsubscribe an invalid session
+		//
 		assertFalse(sessionController.isClientSubscribed(clientCtx));
```
Comment (edgarmueller) [inline com.eclipsesource.modelserver.emf/src/test/java/com/eclipsesource/modelserver/emf/common/SessionControllerTest.java]: Move to own test here as well.

File: com.eclipsesource.modelserver.emf/src/test/java/com/eclipsesource/modelserver/emf/common/SessionControllerTest.java [hunk 5]
```diff
@@ -59,6 +59,8 @@
 	private CommandCodec commandCodec;
 	@Mock
 	private WsContext clientCtx;
+	@Mock
+	private WsContext clientCtx2;
```
Comment (edgarmueller) [inline com.eclipsesource.modelserver.emf/src/test/java/com/eclipsesource/modelserver/emf/common/SessionControllerTest.java]: Could we rename this? I think it's only used in the context of unsubscribing, so maybe something related to that?

**Pull request merged**